### PR TITLE
Feature/remove active scope

### PR DESCRIPTION
**Pull request by bodyangug**

### CODE REVIEW

File: test2code-common/src/commonMain/kotlin/com/epam/drill/plugins/test2code/common/api/Model.kt [hunk 1]
```diff
@@ -95,16 +95,37 @@ data class AgentSessionTestsPayload(val sessionId: String, val tests: List<Strin
...
+        if (testId != other.testId) return false
 
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = id?.hashCode() ?: 0
+        result = 31 * result + className.hashCode()
+        result = 31 * result + (sessionId?.hashCode() ?: 0)
+        result = 31 * result + testId.hashCode()
```
Comment (RomanDavlyatshin) [inline test2code-common/src/commonMain/kotlin/com/epam/drill/plugins/test2code/common/api/Model.kt:128]: - reason for including `sessonId` and `testId`
- reason for using magic number `31`

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

File: dsm/src/main/kotlin/com/epam/dsm/CollectionStoring.kt [hunk 2]
```diff
@@ -29,7 +29,7 @@ import kotlin.reflect.*
 /**
  * File is needed to don't keep a huge collection in memory
  */
-fun <T : Any?> storeCollection(
+inline fun <reified T : Any?> storeCollection(
```
[the remark this answers is not in the document]
Comment (bodyangug) [reply dsm/src/main/kotlin/com/epam/dsm/CollectionStoring.kt:32]: In Kotlin, `reified` preserves information about the actual type of `T` at runtime. This allows you to use `T` as a regular class, defining its corresponding functions and properties.

File: test2code-common/src/commonMain/kotlin/com/epam/drill/plugins/test2code/common/api/Model.kt (hunk 1 above)
Comment (bodyangug) [reply test2code-common/src/commonMain/kotlin/com/epam/drill/plugins/test2code/common/api/Model.kt:128]: equals and hashCode method were autogenerated by IDEA

File: test2code-api/src/commonMain/kotlin/com/epam/drill/plugins/test2code/api/Model.kt [hunk 3]
```diff
@@ -142,6 +142,7 @@ enum class TestResult {
     UNKNOWN
 }
 
+//TODO remove Scope related payloads
```
[the remark this answers is not in the document]
Comment (bodyangug) [reply test2code-api/src/commonMain/kotlin/com/epam/drill/plugins/test2code/api/Model.kt:145]: It remains for backward compatibility.

File: dsm/src/main/kotlin/com/epam/dsm/util/Util.kt [hunk 4]
```diff
@@ -32,7 +33,10 @@ val json = Json {
     encodeDefaults = true
 }
 
-inline fun <reified T : Any> T.id(): String {
+inline fun <reified T : Any> T.id(uuidIfNull: Boolean = false): String {
```
[the remark this answers is not in the document]
Comment (bodyangug) [reply dsm/src/main/kotlin/com/epam/dsm/util/Util.kt:36]: for `true` case  - https://github.com/Drill4J/lib-jvm-shared/blob/7cb8df0396fff7d8dbccd20580ec3110cb5211b7/dsm/src/main/kotlin/com/epam/dsm/CollectionStoring.kt#L60C1-L60C1
for `false` case - https://github.com/Drill4J/lib-jvm-shared/blob/7cb8df0396fff7d8dbccd20580ec3110cb5211b7/dsm/src/main/kotlin/com/epam/dsm/Core.kt#L219C5-L219C5

File: dsm/src/main/kotlin/com/epam/dsm/CollectionStoring.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (RomanDavlyatshin) [reply dsm/src/main/kotlin/com/epam/dsm/CollectionStoring.kt:32]: Spare me that ChatGPT explanation 😄 I know what `reified` does, I just don't get how it is useful here.
Type information is not accessed in any way in `storeCollection`'s code

[the remark this answers is not in the document]
Comment (RomanDavlyatshin) [reply dsm/src/main/kotlin/com/epam/dsm/CollectionStoring.kt:32]: it is required to call `.id` on generic-ly typed var

File: test2code-api/src/commonMain/kotlin/com/epam/drill/plugins/test2code/api/Model.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (RomanDavlyatshin) [reply test2code-api/src/commonMain/kotlin/com/epam/drill/plugins/test2code/api/Model.kt:145]: TODO - create separate task to remove those

**Review by RomanDavlyatshin — APPROVED** [review]

File: test2code-api/src/commonMain/kotlin/com/epam/drill/plugins/test2code/api/Model.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (RomanDavlyatshin) [reply test2code-api/src/commonMain/kotlin/com/epam/drill/plugins/test2code/api/Model.kt:145]: https://github.com/Drill4J/lib-jvm-shared/pull/26

**Pull request merged**